### PR TITLE
dm: delete file ignore blob file not exist

### PR DESCRIPTION
**Pull request by River2000i**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/11912

### What is changed and how it works?
delete file in azure ignore blob file not exist error

#### Tests 

 - Manual test (add detailed scripts or steps below)
create task with azure dir
image
1. target dir is empty dir
image
backup success
image

2. target dir is not empty dir
image
backup success
image

3. target dir contain sub dir
image
backup success
image
4. run dump and load task success
```log
dmctl test cmd: "query-status test-load" with max 10 retries
got=1 expected=1
dmctl test cmd: "query-status test-load" with max 100 retries
got=0 expected=1
command: query-status test-load "stage": "Finished" count: 0 != expected: 1, failed the 0-th time, will retry again
got=1 expected=1
check diff successfully
```
### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: dm/pkg/storage/utils.go [hunk 1]
```diff
@@ -142,15 +143,15 @@ func RemoveAll(ctx context.Context, dir string, storage bstorage.ExternalStorage
 
 	err = storage.WalkDir(ctx, &bstorage.WalkOption{}, func(filePath string, size int64) error {
 		err2 := storage.DeleteFile(ctx, filePath)
-		if errors.Cause(err2) == gstorage.ErrObjectNotExist {
+		if errors.Cause(err2) == gstorage.ErrObjectNotExist || bloberror.HasCode(err2, bloberror.BlobNotFound) {
```
Comment (Benjamin2037) [inline dm/pkg/storage/utils.go:146]: Can we only skip this error when this is az blob?  For other object storage, we can not skip this error.

Comment (River2000i) [reply dm/pkg/storage/utils.go:146]: sure, we can ignore the corresponding errors by backend type.

**Comment by OliverS929** [comment]
LGTM

**Review by lance6716 — APPROVED** [review]

**Comment by Benjamin2037** [comment]
Can we add test case for this pr？

File: dm/pkg/storage/utils.go [hunk 2]
```diff
@@ -133,25 +135,36 @@ func CollectDirFiles(ctx context.Context, dir string, storage bstorage.ExternalS
...
 	})
 	if err == nil {
 		err = storage.DeleteFile(ctx, "")
-		if errors.Cause(err) == gstorage.ErrObjectNotExist {
-			// ignore not exist error when we delete files
+		// ignore not exist error when we delete files and backend is gcs/azure blob storage
+		if backend.GetGcs() != nil && errors.Cause(err) == gstorage.ErrObjectNotExist {
+			return nil
+		}
+		if backend.GetAzureBlobStorage() != nil && bloberror.HasCode(err, bloberror.BlobNotFound) {
```
Comment (D3Hunter) [inline dm/pkg/storage/utils.go:167]: `backend` might be nil

Comment (River2000i) [reply dm/pkg/storage/utils.go]: fix in https://github.com/pingcap/tiflow/pull/11913/commits/4560475723b25a613e4dd924c97db765d207e9e4

**Review by D3Hunter — APPROVED** [review]

**Comment by River2000i** [comment]
> Can we add test case for this pr？

Maybe we can add test case in internal path?

**Pull request merged**